### PR TITLE
[openvino] 2024.0.0 release

### PR DESCRIPTION
**Pull request by RaguzovaTatyana**
Specify version: openvino/2024.0.0

### CODE REVIEW

**Comment by MonicaLiu0311** [comment]
Please get [failure logs here](https://dev.azure.com/vcpkg/public/_build/results?buildId=99901&view=results):
```
Building openvino[auto,auto-batch,core,cpu,gpu,hetero,ir,onnx,paddle,pytorch,tensorflow,tensorflow-lite]:x64-windows@2024.0.0...
-- Downloading https://github.com/openvinotoolkit/openvino/pull/22011.diff?full_index=1 -> openvino-pr-22011.patch...
-- Downloading https://github.com/openvinotoolkit/openvino/archive/34caeefd07800b59065345d651949efbe8ab6649.tar.gz -> openvinotoolkit-openvino-34caeefd07800b59065345d651949efbe8ab6649.tar.gz...
-- Extracting source D:/downloads/openvinotoolkit-openvino-34caeefd07800b59065345d651949efbe8ab6649.tar.gz
-- Applying patch 001-disable-tools.patch
-- Applying patch D:/downloads/openvino-pr-22011.patch
CMake Error at scripts/cmake/z_vcpkg_apply_patches.cmake:34 (message):
  Applying patch failed: Checking patch
  src/plugins/intel_gpu/src/runtime/ocl/ocl_device.cpp...

  error: while searching for:

  } // namespace





  ocl_device::ocl_device(const cl::Device dev, const cl::Context& ctx, const
  cl_platform_id platform)

  : _context(ctx)

  , _device(dev)

  , _platform(platform)



  error: patch failed:
  src/plugins/intel_gpu/src/runtime/ocl/ocl_device.cpp:299

  error: src/plugins/intel_gpu/src/runtime/ocl/ocl_device.cpp: patch does not

  error: while searching for:

              for (auto& device : devices) {
                  if (!does_device_match_config(device))
                      continue;
                  supported_devices.emplace_back(std::make_shared<ocl_device>(device, cl::Context(device), id));
              }
          } catch (std::exception& ex) {
              GPU_DEBUG_LOG << "Devices query/creation failed for " << platform.getInfo<CL_PLATFORM_NAME>() << ": " << ex.what() << std::endl;

  

  error: patch failed:
  src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp:207

  error: src/plugins/intel_gpu/src/runtime/ocl/ocl_device_detector.cpp: patch
  does not apply

  Checking patch thirdparty/ocl/cl_headers...

  Checking patch thirdparty/ocl/clhpp_headers...

  Checking patch thirdparty/ocl/icd_loader...

Call Stack (most recent call first):
  scripts/cmake/vcpkg_extract_source_archive.cmake:147 (z_vcpkg_apply_patches)
  scripts/cmake/vcpkg_extract_source_archive_ex.cmake:8 (vcpkg_extract_source_archive)
  scripts/cmake/vcpkg_from_github.cmake:113 (vcpkg_extract_source_archive_ex)
  ports/openvino/portfile.cmake:7 (vcpkg_from_github)
  scripts/ports.cmake:170 (include)


error: building openvino:x64-windows failed with: BUILD_FAILED
Elapsed time to handle openvino:x64-windows: 10 s
Total install time: 11 s
```

**Comment by MonicaLiu0311** [comment]
If you're ready, please revert to "ready for review".

**Comment by RaguzovaTatyana** [comment]
@MonicaLiu0311 
Ok, I'm waiting for release day and tag 2024.0.0 in the openvino repository for now.

**Comment by RaguzovaTatyana** [comment]
@MonicaLiu0311 Could you please review this PR?

File: ports/openvino/portfile.cmake [hunk 1]
```diff
@@ -1,20 +1,11 @@
...
-    FILENAME "openvino-pr-22011.patch"
-    SHA512 f25de88db953a825d0a9a2f8e78664aba44eda2fecfed28ba52045c82aa5f586e42deb51cf23c47553e20c64942510500caca6e904d27ab0782270cd3047e953
-)
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO openvinotoolkit/openvino
     REF "${VERSION}"
-    SHA512 d31428a02e6869e367c00a714c5599cf84fc9839376dfc4511d2abdb4aac991fc58331318102cdc248f568ca8df0bdc7f65a6744ef2dfa52d85a60a1ec77aae9
+    SHA512 b5ccf6fc75cb2804930f0f1ba30755ea0b1d36b58fc1b3437cfefebb69a9bf33c785cd45ab8c767c9af27fbd79956e7956d89efe2c408a9e328f7d9d305d6dfe
```
Comment (MonicaLiu0311) [inline ports/openvino/portfile.cmake:5]:
```
Installing 27/27 openvino[auto,auto-batch,core,cpu,gpu,hetero,ir,onnx,paddle,pytorch,tensorflow,tensorflow-lite]:x64-windows@2024.0.0...
Building openvino[auto,auto-batch,core,cpu,gpu,hetero,ir,onnx,paddle,pytorch,tensorflow,tensorflow-lite]:x64-windows@2024.0.0...
-- Downloading https://github.com/openvinotoolkit/openvino/archive/2024.0.0.tar.gz -> openvinotoolkit-openvino-2024.0.0.tar.gz...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Trying to load bundleconfig from G:\openvino\vcpkg-bundle.json
[DEBUG] Failed to open: G:\openvino\vcpkg-bundle.json
[DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
[DEBUG] Metrics enabled.
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Feature flag 'dependencygraph' unset
Downloading https://github.com/openvinotoolkit/openvino/archive/2024.0.0.tar.gz
[DEBUG] Trying to hash G:\openvino\downloads\openvinotoolkit-openvino-2024.0.0.tar.gz.6096.part
[DEBUG] G:\openvino\downloads\openvinotoolkit-openvino-2024.0.0.tar.gz.6096.part has hash 89c60aa8c73d91003f9fdd4e8f93213b98164ecdcd4a4264ba93352473666a290d1dd55b61cf9c5d62b0cef429f989bb31f5d4bfa67cf6d97dfc2abca45d4186
error: Failed to download from mirror set
error: File does not have the expected hash:
url: https://github.com/openvinotoolkit/openvino/archive/2024.0.0.tar.gz
File: G:\openvino\downloads\openvinotoolkit-openvino-2024.0.0.tar.gz.6096.part
Expected hash: b5ccf6fc75cb2804930f0f1ba30755ea0b1d36b58fc1b3437cfefebb69a9bf33c785cd45ab8c767c9af27fbd79956e7956d89efe2c408a9e328f7d9d305d6dfe
Actual hash: 89c60aa8
```

Comment (RaguzovaTatyana) [reply ports/openvino/portfile.cmake]: Thank you, fixed

**Comment by MonicaLiu0311** [comment]
All features are tested successfully in the following triplet:
```
x64-windows
x64-windows-static
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**